### PR TITLE
Change inherit_from file loading

### PR DESCRIPTION
**Pull request by ricardoperez**
# Description

Adding the possibility to load a decision file from a specific gem and
allowing adition of authorization http header from remote files.

## Inherit from a gem
```yml
---
- - :inherit_from
  -
    gem: gem_name
    path: 'doc/dependency_decisions.yml'
```

### Add Authorization Bearer token
```yml
---
- - :inherit_from
  -
    url: https://raw.githubusercontent.com/forward3d/repo/master/doc/dependency_decisions.yml
    token: $ENV_VAR_WITH_TOKEN
```

### CODE REVIEW

File: spec/lib/license_finder/decisions_spec.rb [hunk 1]
```diff
@@ -291,7 +291,35 @@ module LicenseFinder
 
       it 'inheritates rules from remote decision file' do
         stub_request(:get, 'https://example.com/config/inherit.yml').to_return(status: 200, body: yml, headers: {})
-        decisions = subject.inherit_from('https://example.com/config/inherit.yml')
```
Comment (PatrykF3D) [inline spec/lib/license_finder/decisions_spec.rb:294]: I think the previous functionality should not be removed.

File: lib/license_finder/decisions.rb [hunk 2]
```diff
@@ -183,19 +183,35 @@ def unname_project(txn = {})
       self
     end
 
-    def inherit_from(filepath)
+    def inherit_from(filepath_info)
       decisions =
-        if filepath =~ %r{^https?://}
```
Comment (PatrykF3D) [inline lib/license_finder/decisions.rb:188]: I think the previous functionality should not be removed.

Comment (ricardoperez) [reply lib/license_finder/decisions.rb:188]: done.

File: spec/lib/license_finder/decisions_spec.rb (hunk 1 above)
Comment (ricardoperez) [reply spec/lib/license_finder/decisions_spec.rb:294]: Done.

File: lib/license_finder/decisions.rb (hunk 2 above)
Comment (PatrykF3D) [reply lib/license_finder/decisions.rb:188]: hash needs to go first, 

ruby 2.7:
```
2.7.0 :001 > h = {}
2.7.0 :002 > h =~ /test/
(irb):2: warning: deprecated Object#=~ is called on Hash; it always returns nil
 => nil 
2.7.0 :003 > 
```

Comment (ricardoperez) [reply lib/license_finder/decisions.rb:188]: Thanks for spotting this.

Comment (ricardoperez) [reply lib/license_finder/decisions.rb:188]: fixed.

**Pull request merged**